src/main/java/hsien/Hsien.java

```diff
@@ -0,0 +1,260 @@
+        int index = Integer.parseInt(desc);
+        boolean isMarking = command.equals("mark");
+
+        // Perform marking or unmarking
+        if (isMarking) {
+            messages.get(index - 1).mark();
+            System.out.println("You marked " + index + " as marked");
+        } else {
+            messages.get(index - 1).unmark();
+            System.out.println("You marked " + index + " as unmarked");
+        }
+
+        // Print status of the task
+        System.out.println(messages.get(index - 1).getStatusDescription());
+    }
+
+    public static void deleteTask(ArrayList<Task> messages, int index) {
+        System.out.println("Noted I have removed this task");
+        System.out.println(messages.get(index).getStatusDescription());
+        messages.remove(index);
+        System.out.printf("Now you have %d tasks in the list\n", messages.size());
+    }
+
+    public static void main(String[] args) {
+        // Initialise variables
+        List<String> validCommands = Arrays.asList("bye", "list", "mark", "unmark", "delete", "todo", "deadline", "event");
+        ArrayList<Task> messages = new ArrayList<>();
+        Scanner in = new Scanner(System.in);
+        boolean isRunning = true;
+
+        welcomeMessage();
+        readFile(messages);
+        printLine();
+
+        while (isRunning) {
+            printCommands(validCommands);
+            String input = in.nextLine();
+
+            String[] processedCommand = processCommand(input);
+            // Extract out the exact command
+            String command = processedCommand[0];
+            String desc = processedCommand[1];
+
+            // Check if valid command
+            try {
+                if (!validCommands.contains(command)) {
+                    throw new HsienException();
+                }
+            } catch (HsienException e) {
+                System.out.println("Please enter a valid command from the list! ");
+                printLine();
+                continue;
+            }
+
+            // Handle command
+            if (command.equals("bye")) {
+                // Exit
+                System.out.println("Have a good day! Bye!");
+                isRunning = false;
+                continue;
+            } else if (command.equals("list")) {
+                printList(messages);
+            } else if (command.startsWith("mark") || command.startsWith("unmark")) {
+                try {
+                    handleMark(command, desc, messages);
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Index out of bounds");
+                }
+            } else if (command.equals("delete")) {
+                try {
+                    deleteTask(messages, Integer.parseInt(desc)-1);
+                } catch (NumberFormatException e) {
+                    System.out.println("Invalid number format");
+                }
+            } else {
+                addTask(command, desc, messages);
```

Comment:
Consider switch case to increase readability, and to prevent arrow head structure